crates/storage/db/src/tables/mod.rs

```diff
@@ -101,10 +101,11 @@ pub trait TableViewer<R> {
 /// General trait for defining the set of tables
 /// Used to initialize database
 pub trait TableSet {
-    /// Returns all the table names in the database.
-    fn table_names(&self) -> Vec<&'static str>;
-    /// Returns `true` if the table at the given index is a `DUPSORT` table.
-    fn is_dupsort(&self, idx: usize) -> bool;
+    /// The iterator type for the tables.
+    type TableIter: Iterator<Item = Box<dyn TableMetadata>>;
+
+    /// Returns an iterator over the tables.
+    fn tables() -> Self::TableIter;
```

Comment:
this can just return Boxed iter of Table instead, no need for the associated type

Reply:
you mean directly using `Iterator<Item = Box<dyn TableMetadata>>` as return type for the `tables` function and remove the `TableIter` associated type?